pandas/tests/extension/test_arrow.py

```diff
@@ -77,6 +78,14 @@ def data(dtype):
         data = [1, 0] * 4 + [None] + [-2, -1] * 44 + [None] + [1, 99]
     elif pa.types.is_unsigned_integer(pa_dtype):
         data = [1, 0] * 4 + [None] + [2, 1] * 44 + [None] + [1, 99]
+    elif pa.types.is_decimal(pa_dtype):
+        data = (
+            [Decimal("1"), Decimal("0.0")] * 4
+            + [None]
+            + [Decimal("-2.0"), Decimal("-1.0")] * 44
+            + [None]
+            + [Decimal("0.5"), Decimal("33.123")]
+        )
```

Comment:
looks like if you pass Decimal("nan") to pyarrow it raises.  is that going to be relevant for us?

Reply:
Looks like we coerce to NA

```
In [6]: pd.Series([decimal.Decimal("nan")], dtype=pd.ArrowDtype(pa.decimal128(4, 1)))
Out[6]:
0    <NA>
dtype: decimal128(4, 1)[pyarrow]
```

Which I guess is fuzzy with the NA vs nan debate